src/chart/bar/BarView.js

```diff
@@ -182,13 +187,19 @@ export default echarts.extendChartView({
                 var layout = getLayout[coord.type](data, newIndex, itemModel);
 
                 if (drawBackground) {
-                    var bgEl = oldBgEls[oldIndex];
-                    bgEl.useStyle(backgroundModel.getBarItemStyle());
-                    // Only cartesian2d support borderRadius.
-                    if (coord.type === 'cartesian2d') {
-                        bgEl.setShape('r', barBorderRadius);
+                    var bgEl;
+                    if (oldBgEls.length === 0) {
+                        bgEl = createBackground(oldIndex);
+                    }
+                    else {
+                        var bgEl = oldBgEls[oldIndex];
```

Comment:
```suggestion
                        bgEl = oldBgEls[oldIndex];
```
The `var` should be removed, otherwise, it may lead to an error after confusion and compression.

Reply:
Uhhhh, missed that in the review.

Reply:
Uhhhh, missed that in the review.

Reply:
Since this PR has already been merged and closed, I cannot append commits on it. I've created a new one #13085 to apply this change. Sorry for this mistake.